From a2f32cec773113a9d0a80f0b20947da9bc29f3bd Mon Sep 17 00:00:00 2001
From: Simon Casey <simon501098c@gmail.com>
Date: Wed, 2 Feb 2022 20:17:21 +0100
Subject: [PATCH] Update
 902-arm64-provide-dma-cache-routines-with-same-API-as-32.patch for kernel
 5.15

---
 arch/arm64/include/asm/cacheflush.h | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/arm64/include/asm/cacheflush.h
+++ b/arch/arm64/include/asm/cacheflush.h
@@ -110,6 +110,15 @@ static inline void flush_icache_range(un
 extern void __dma_map_area(const void *, size_t, int);
 extern void __dma_unmap_area(const void *, size_t, int);
 extern void __dma_flush_area(const void *, size_t);
+extern void __dma_inv_area(const void *start, const void *end);
+extern void __dma_clean_area(const void *start, const void *end);
+
+#define dmac_flush_range(start, end) \
+	__dma_flush_area(start, (void *)(end) - (void *)(start))
+#define dmac_inv_range(start, end) \
+	__dma_inv_area(start, end)
+#define dmac_clean_range(start, end) \
+	__dma_clean_area(start, end)
 
 /*
  * Copy user data from/to a page which is mapped into a different
